shadowaead/packet.go

```diff
@@ -27,6 +29,7 @@ func Pack(dst, plaintext []byte, ciph Cipher) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	saltfilter.Add(salt)
```

Comment:
So here's the problem with the package-level global variable. What if the caller initiates more than one connection? I think the Bloom filter should be coupled with the Cipher, as it is fine to use the same salt if two different keys are used.

Reply:
In my mind, expose the package-level global variable/function is to make the filter instance shared between `aead` and `stream` packages. And why do this instead of coupled with cipher? Coz the chance of repeating salt is too small to care and even it happens the retry machinery will handle this. So I choose to use a global variable.

Reply:
I see. Actually what you said does make sense in this case.

Reply:
Decoupling with `saltfilter` package maybe this is a good way(I thought)